scripts/emoji_codegen.py

```diff
@@ -68,16 +105,67 @@ def generate_qml_list(**kwargs):
         code, qualification, charAndName = segments
 
         # skip unqualified versions of same unicode
-        if qualification == 'unqualified':
-            continue
-
-        if qualification == 'component':
+        if qualification != 'fully-qualified' and qualification != 'component' :
             continue
+        
 
         char, name = re.match(r'^(\S+) E\d+\.\d+ (.*)$', charAndName).groups()
-
-        categories[current_category].append(Emoji(code, name))
+        shortname = name
+        
+        # discard skin tone variants for sanity
+        # __contains__ is so stupid i hate prototype languages
+        if name.__contains__("skin tone") and qualification != 'component': 
+            continue
+        if qualification == 'component' and not name.__contains__("skin tone"): 
+            continue
+        #TODO: Handle skintone modifiers in a sane way
+        if shortname in shortcodeDict: 
+            shortname = shortcodeDict[shortname]
+        else:
+            shortname = shortname.lower()
+            if shortname.endswith(' (blood type)'): 
+                shortname = shortname[:-13]
+            if shortname.endswith(': red hair'): 
+                shortname = "red_haired_" + shortname[:-10]
+            if shortname.endswith(': curly hair'): 
+                shortname = "curly_haired_" + shortname[:-12]
+            if shortname.endswith(': white hair'): 
+                shortname = "white_haried_" + shortname[:-12]
+            if shortname.endswith(': bald'): 
+                shortname = "bald_" + shortname[:-6]
+            if shortname.endswith(': beard'): 
+                shortname = "bearded_" + shortname[:-7]
+            if shortname.endswith(' face'): 
+                shortname = shortname[:-5]
+            if shortname.endswith(' button'): 
+                shortname = shortname[:-7] 
+            if shortname.endswith(' banknote'): 
+                shortname = shortname[:-9]
+            keycapmtch = re.match(r'^keycap: (.+)$', shortname)
+            if keycapmtch: 
+                keycapthing, = keycapmtch.groups()
+                type(keycapthing)
+                num_name = humanize_keypad(keycapthing) 
```

Comment:
Why would you want to type out the number instead of just using your number key?